integration-tests/e2e/emaildomain.create.cy.js

```diff
@@ -0,0 +1,81 @@
+    cy.task('stubSignIn', { roles: [{ roleCode: 'MAINTAIN_EMAIL_DOMAINS' }] })
+    cy.signIn()
+    cy.task('stubAllEmailDomains', {})
+    MenuPage.verifyOnPage().viewEmailDomainListing()
+    const emailDomainListingPage = EmailDomainListingPage.verifyOnPage()
+    emailDomainListingPage.navigateToCreateEmailDomainPageButton().click()
+    const emailDomainCreationPage = EmailDomainCreationPage.verifyOnPage()
+    emailDomainCreationPage.createEmailDomainPage('Domain1', '')
+    emailDomainCreationPage.errorSummary().should('contain.text', 'Enter a domain description')
+  })
+
+  it('Should display error and prevent form submission, when email domain name is blank', () => {
+    cy.task('stubSignIn', { roles: [{ roleCode: 'MAINTAIN_EMAIL_DOMAINS' }] })
+    cy.signIn()
+    cy.task('stubAllEmailDomains', {})
+    MenuPage.verifyOnPage().viewEmailDomainListing()
+    const emailDomainListingPage = EmailDomainListingPage.verifyOnPage()
+    emailDomainListingPage.navigateToCreateEmailDomainPageButton().click()
+    const emailDomainCreationPage = EmailDomainCreationPage.verifyOnPage()
+    emailDomainCreationPage.createEmailDomainPage('', 'Domain1Description')
+    emailDomainCreationPage.errorSummary().should('contain.text', 'Enter a domain name')
+  })
+
+  it('Should display error and prevent form submission, when both email domain name and description is blank', () => {
+    cy.task('stubSignIn', { roles: [{ roleCode: 'MAINTAIN_EMAIL_DOMAINS' }] })
+    cy.signIn()
+    cy.task('stubAllEmailDomains', {})
+    MenuPage.verifyOnPage().viewEmailDomainListing()
+
+    const emailDomainListingPage = EmailDomainListingPage.verifyOnPage()
+    emailDomainListingPage.navigateToCreateEmailDomainPageButton().click()
+    const emailDomainCreationPage = EmailDomainCreationPage.verifyOnPage()
+    emailDomainCreationPage.createEmailDomainPage('', '')
+    emailDomainCreationPage.errorSummary().should('contain.text', 'Enter a domain description')
+    emailDomainCreationPage.errorSummary().should('contain.text', 'Enter a domain name')
+  })
+
+  it('Should navigate to email domain listing, on-clicking the  cancel create email domain button', () => {
+    cy.task('stubSignIn', { roles: [{ roleCode: 'MAINTAIN_EMAIL_DOMAINS' }] })
+    cy.signIn()
+    cy.task('stubAllEmailDomains', {})
+    MenuPage.verifyOnPage().viewEmailDomainListing()
+
+    const emailDomainListingPage = EmailDomainListingPage.verifyOnPage()
+    emailDomainListingPage.navigateToCreateEmailDomainPageButton().click()
+
+    const emailDomainCreationPage = EmailDomainCreationPage.verifyOnPage()
+    emailDomainCreationPage.cancel()
+    emailDomainListingPage.emailDomainListingTableRows().should('have.length', 3)
+    emailDomainListingPage.emailDomainListingTableRows().eq(0).should('include.text', 'DOMAIN1')
+    emailDomainListingPage.emailDomainListingTableRows().eq(0).should('include.text', 'DOMAIN 1 DESCRIPTION')
+  })
```

Comment:
are we missing a CSRF token check?

Reply:
CSRF Defo. 
The disallowing of direct access when user does not have role is already covered in the email domain listing which is the entry point, but can replicate for create email domain.